kernel/src/main/java/org/kframework/kompile/KompileUsageFormatter.java

```diff
@@ -0,0 +1,397 @@
+// Copyright (c) K Team. All Rights Reserved.
+package org.kframework.kompile;
+
+import com.beust.jcommander.*;
+import com.beust.jcommander.internal.Lists;
+
+import java.util.*;
+import java.util.ResourceBundle;
+
+/**
+ * Customized `usage` for kompile based on DefaultUsageFormatter.
+ * The only changes are related to printing hidden options when
+ * --help-hidden is provided.
```

Comment:
Is the rest of the code taken from an example implementation?

Reply:
This is the default implementation for the JCommander library. 
I took the code from here: https://github.com/cbeust/jcommander/blob/master/src/main/java/com/beust/jcommander/DefaultUsageFormatter.java
and modified the parts related to printing hidden attributes.
One thing that I'm not sure is how to handle the license. The original file is Apache 2.0.

Reply:
Seems fine to do this, but you'll need to add an attribution notice I think: https://snyk.io/learn/apache-license/#apache-license-permissions

Reply:
Added a notice comment. Please have another look.